### PR TITLE
Add OWIN docs

### PR DESCRIPTION
**Pull request by JamesNK**
Addresses https://github.com/dotnet/aspnetcore/issues/27538

I've never used OWIN before so some comments might need improvement.

### CODE REVIEW

File: src/Http/Owin/src/OwinExtensions.cs [hunk 1]
```diff
@@ -77,11 +91,22 @@ public static IApplicationBuilder UseOwin(this IApplicationBuilder builder, Acti
             return builder;
         }
 
+        /// <summary>
+        /// Create an <see cref="IApplicationBuilder"/> for an OWIN pipeline.
```
Comment (pranavkm) [inline src/Http/Owin/src/OwinExtensions.cs:95]:
```suggestion
        /// Creates an <see cref="IApplicationBuilder"/> for an OWIN pipeline.
```

File: src/Http/Owin/src/OwinExtensions.cs [hunk 2]
```diff
@@ -77,11 +91,22 @@ public static IApplicationBuilder UseOwin(this IApplicationBuilder builder, Acti
...
+        /// </summary>
+        /// <param name="app">The OWIN pipeline.</param>
+        /// <returns>An <see cref="IApplicationBuilder"/></returns>
         public static IApplicationBuilder UseBuilder(this AddMiddleware app)
         {
             return app.UseBuilder(serviceProvider: null);
         }
 
+        /// <summary>
+        /// Create an <see cref="IApplicationBuilder"/> for an OWIN pipeline.
```
Comment (pranavkm) [inline src/Http/Owin/src/OwinExtensions.cs:105]:
```suggestion
        /// Creates an <see cref="IApplicationBuilder"/> for an OWIN pipeline.
```

File: src/Http/Owin/src/OwinExtensions.cs [hunk 3]
```diff
@@ -142,11 +167,24 @@ private static CreateMiddleware CreateMiddlewareFactory(Func<RequestDelegate, Re
             };
         }
 
+        /// <summary>
+        /// Create an <see cref="IApplicationBuilder"/> for an OWIN pipeline.
```
Comment (pranavkm) [inline src/Http/Owin/src/OwinExtensions.cs:171]:
```suggestion
        /// Creates an <see cref="IApplicationBuilder"/> for an OWIN pipeline.
```

File: src/Http/Owin/src/OwinExtensions.cs [hunk 4]
```diff
@@ -142,11 +167,24 @@ private static CreateMiddleware CreateMiddlewareFactory(Func<RequestDelegate, Re
...
+        /// <param name="app">The OWIN pipeline.</param>
+        /// <param name="pipeline">A delegate used to configure a middleware pipeline.</param>
+        /// <returns>An <see cref="IApplicationBuilder"/>.</returns>
         public static AddMiddleware UseBuilder(this AddMiddleware app, Action<IApplicationBuilder> pipeline)
         {
             return app.UseBuilder(pipeline, serviceProvider: null);
         }
 
+        /// <summary>
+        /// Create an <see cref="IApplicationBuilder"/> for an OWIN pipeline.
```
Comment (pranavkm) [inline src/Http/Owin/src/OwinExtensions.cs:182]:
```suggestion
        /// Creates an <see cref="IApplicationBuilder"/> for an OWIN pipeline.
```

**Review by pranavkm — APPROVED** [review]

File: src/Http/Owin/src/OwinEnvironment.cs [hunk 5]
```diff
@@ -290,27 +301,63 @@ IEnumerator IEnumerable.GetEnumerator()
...
+            /// </summary>
+            /// <param name="featureInterface">The feature interface type.</param>
+            /// <param name="getter">Value getter.</param>
             public FeatureMap(Type featureInterface, Func<object, object> getter)
                 : this(featureInterface, getter, defaultFactory: null)
             {
             }
+
+            /// <summary>
+            /// Create a <see cref="FeatureMap"/> for the specified feature interface type.
```
Comment (JamesNK) [inline src/Http/Owin/src/OwinEnvironment.cs:320]:
```suggestion
            /// Initializes a new instance of <see cref="FeatureMap"/> for the specified feature interface type.
```

File: src/Http/Owin/src/OwinEnvironment.cs [hunk 6]
```diff
@@ -290,27 +301,63 @@ IEnumerator IEnumerable.GetEnumerator()
...
+            /// <param name="featureInterface">The feature interface type.</param>
+            /// <param name="getter">Value getter delegate.</param>
+            /// <param name="defaultFactory">Default value factory delegate.</param>
             public FeatureMap(Type featureInterface, Func<object, object> getter, Func<object> defaultFactory)
                 : this(featureInterface, getter, defaultFactory, setter: null)
             {
             }
 
+            /// <summary>
+            /// Create a <see cref="FeatureMap"/> for the specified feature interface type.
```
Comment (JamesNK) [inline src/Http/Owin/src/OwinEnvironment.cs:331]:
```suggestion
            /// Initializes a new instance of <see cref="FeatureMap"/> for the specified feature interface type.
```

File: src/Http/Owin/src/OwinEnvironment.cs [hunk 7]
```diff
@@ -290,27 +301,63 @@ IEnumerator IEnumerable.GetEnumerator()
...
+            /// <param name="featureInterface">The feature interface type.</param>
+            /// <param name="getter">Value getter delegate.</param>
+            /// <param name="setter">Value setter delegate.</param>
             public FeatureMap(Type featureInterface, Func<object, object> getter, Action<object, object> setter)
                 : this(featureInterface, getter, defaultFactory: null, setter: setter)
             {
             }
 
+            /// <summary>
+            /// Create a <see cref="FeatureMap"/> for the specified feature interface type.
```
Comment (JamesNK) [inline src/Http/Owin/src/OwinEnvironment.cs:342]:
```suggestion
            /// Initializes a new instance of <see cref="FeatureMap"/> for the specified feature interface type.
```

File: src/Http/Owin/src/OwinEnvironment.cs [hunk 8]
```diff
@@ -290,27 +301,63 @@ IEnumerator IEnumerable.GetEnumerator()
...
+            /// <param name="getter">Value getter delegate.</param>
+            /// <param name="defaultFactory">Default value factory delegate.</param>
+            /// <param name="setter">Value setter delegate.</param>
             public FeatureMap(Type featureInterface, Func<object, object> getter, Func<object> defaultFactory, Action<object, object> setter)
                 : this(featureInterface, getter, defaultFactory, setter, featureFactory: null)
             {
             }
 
+            /// <summary>
+            /// Create a <see cref="FeatureMap"/> for the specified feature interface type.
```
Comment (JamesNK) [inline src/Http/Owin/src/OwinEnvironment.cs:354]:
```suggestion
            /// Initializes a new instance of <see cref="FeatureMap"/> for the specified feature interface type.
```

File: src/Http/Owin/src/OwinEnvironment.cs [hunk 9]
```diff
@@ -366,28 +416,59 @@ internal void Set(HttpContext context, object value)
...
         }
 
+        /// <summary>
+        /// Maps OWIN keys to ASP.NET Core features.
+        /// </summary>
+        /// <typeparam name="TFeature">Feature interface type.</typeparam>
         public class FeatureMap<TFeature> : FeatureMap
         {
+            /// <summary>
+            /// Create a <see cref="FeatureMap"/> for the specified feature interface type.
```
Comment (JamesNK) [inline src/Http/Owin/src/OwinEnvironment.cs:426]:
```suggestion
            /// Initializes a new instance of <see cref="FeatureMap"/> for the specified feature interface type.
```

File: src/Http/Owin/src/OwinEnvironment.cs [hunk 10]
```diff
@@ -366,28 +416,59 @@ internal void Set(HttpContext context, object value)
...
+            /// </summary>
+            /// <param name="getter">Value getter.</param>
             public FeatureMap(Func<TFeature, object> getter)
                 : base(typeof(TFeature), feature => getter((TFeature)feature))
             {
             }
 
+            /// <summary>
+            /// Create a <see cref="FeatureMap"/> for the specified feature interface type.
```
Comment (JamesNK) [inline src/Http/Owin/src/OwinEnvironment.cs:435]:
```suggestion
            /// Initializes a new instance of <see cref="FeatureMap"/> for the specified feature interface type.
```

File: src/Http/Owin/src/OwinEnvironment.cs [hunk 11]
```diff
@@ -366,28 +416,59 @@ internal void Set(HttpContext context, object value)
...
+            /// </summary>
+            /// <param name="getter">Value getter delegate.</param>
+            /// <param name="defaultFactory">Default value factory delegate.</param>
             public FeatureMap(Func<TFeature, object> getter, Func<object> defaultFactory)
                 : base(typeof(TFeature), feature => getter((TFeature)feature), defaultFactory)
             {
             }
 
+            /// <summary>
+            /// Create a <see cref="FeatureMap"/> for the specified feature interface type.
```
Comment (JamesNK) [inline src/Http/Owin/src/OwinEnvironment.cs:445]:
```suggestion
            /// Initializes a new instance of <see cref="FeatureMap"/> for the specified feature interface type.
```

File: src/Http/Owin/src/OwinEnvironment.cs [hunk 12]
```diff
@@ -366,28 +416,59 @@ internal void Set(HttpContext context, object value)
...
+            /// </summary>
+            /// <param name="getter">Value getter delegate.</param>
+            /// <param name="setter">Value setter delegate.</param>
             public FeatureMap(Func<TFeature, object> getter, Action<TFeature, object> setter)
                 : base(typeof(TFeature), feature => getter((TFeature)feature), (feature, value) => setter((TFeature)feature, value))
             {
             }
 
+            /// <summary>
+            /// Create a <see cref="FeatureMap"/> for the specified feature interface type.
```
Comment (JamesNK) [inline src/Http/Owin/src/OwinEnvironment.cs:455]:
```suggestion
            /// Initializes a new instance of <see cref="FeatureMap"/> for the specified feature interface type.
```

File: src/Http/Owin/src/OwinEnvironment.cs [hunk 13]
```diff
@@ -366,28 +416,59 @@ internal void Set(HttpContext context, object value)
...
+            /// <param name="getter">Value getter delegate.</param>
+            /// <param name="defaultFactory">Default value factory delegate.</param>
+            /// <param name="setter">Value setter delegate.</param>
             public FeatureMap(Func<TFeature, object> getter, Func<object> defaultFactory, Action<TFeature, object> setter)
                 : base(typeof(TFeature), feature => getter((TFeature)feature), defaultFactory, (feature, value) => setter((TFeature)feature, value))
             {
             }
 
+            /// <summary>
+            /// Create a <see cref="FeatureMap"/> for the specified feature interface type.
```
Comment (JamesNK) [inline src/Http/Owin/src/OwinEnvironment.cs:466]:
```suggestion
            /// Initializes a new instance of <see cref="FeatureMap"/> for the specified feature interface type.
```

File: src/Http/Owin/src/WebSockets/OwinWebSocketAcceptContext.cs [hunk 14]
```diff
@@ -6,19 +6,30 @@
...
 {
+    /// <summary>
+    /// OWIN WebSocket accept context.
+    /// </summary>
     public class OwinWebSocketAcceptContext : WebSocketAcceptContext
     {
         private IDictionary<string, object> _options;
 
+        /// <summary>
+        /// Create an <see cref="OwinWebSocketAcceptContext"/>.
```
Comment (JamesNK) [inline src/Http/Owin/src/WebSockets/OwinWebSocketAcceptContext.cs:17]:
```suggestion
        /// Initializes a new instance of <see cref="OwinWebSocketAcceptContext"/>.
```

File: src/Http/Owin/src/WebSockets/OwinWebSocketAcceptContext.cs [hunk 15]
```diff
@@ -6,19 +6,30 @@
...
+        /// </summary>
         public OwinWebSocketAcceptContext() : this(new Dictionary<string, object>(1))
         {
         }
 
+        /// <summary>
+        /// Create an <see cref="OwinWebSocketAcceptContext"/>.
```
Comment (JamesNK) [inline src/Http/Owin/src/WebSockets/OwinWebSocketAcceptContext.cs:24]:
```suggestion
        /// Initializes a new instance of <see cref="OwinWebSocketAcceptContext"/>.
```

File: src/Http/Owin/src/WebSockets/OwinWebSocketAdapter.cs [hunk 16]
```diff
@@ -42,6 +45,11 @@ public class OwinWebSocketAdapter : WebSocket
         private WebSocketState _state;
         private string _subProtocol;
 
+        /// <summary>
+        /// Create an <see cref="OwinWebSocketAdapter"/>.
```
Comment (JamesNK) [inline src/Http/Owin/src/WebSockets/OwinWebSocketAdapter.cs:49]:
```suggestion
        /// Initializes a new instance of <see cref="OwinWebSocketAdapter"/>.
```

File: src/Http/Owin/src/WebSockets/WebSocketAcceptAdapter.cs [hunk 17]
```diff
@@ -39,6 +39,11 @@ public class WebSocketAcceptAdapter
         private AppFunc _callback;
         private IDictionary<string, object> _options;
 
+        /// <summary>
+        /// Create a <see cref="WebSocketAcceptAdapter"/> for an OWIN environment.
```
Comment (JamesNK) [inline src/Http/Owin/src/WebSockets/WebSocketAcceptAdapter.cs:43]:
```suggestion
        /// Initializes a new instance of <see cref="WebSocketAcceptAdapter"/> for an OWIN environment.
```

File: src/Http/Owin/src/OwinEnvironment.cs [hunk 18]
```diff
@@ -28,11 +28,18 @@ namespace Microsoft.AspNetCore.Owin
             Task<WebSocket>
         >;
 
+    /// <summary>
+    /// A strongly-typed wrapper of an OWIN environment.
```
Comment (Tratcher) [inline src/Http/Owin/src/OwinEnvironment.cs:32]: It's actually the other way around.
```suggestion
    /// A loosely-typed OWIN environment wrapper over an <see cref="HttpContext"/>.
```

**Review by Tratcher — APPROVED** [review]

**Pull request merged**